src/lib/mina_networking/mina_networking.ml

```diff
@@ -1482,21 +1482,25 @@ let broadcast_state t state =
   Mina_metrics.(Gauge.inc_one Network.new_state_broadcasted) ;
   Gossip_net.Any.broadcast_state t.gossip_net msg
 
-let broadcast_transaction_pool_diff t diff =
-  log_gossip t.logger (Gossip_net.Message.Transaction_pool_diff diff)
+let broadcast_transaction_pool_diff ?nonce t diff =
+  log_gossip t.logger
+    (Gossip_net.Message.Transaction_pool_diff
+       { message = diff; nonce = Option.value ~default:0 nonce } )
     ~log_msg:(Gossip_transaction_pool_diff { txns = diff }) ;
   Mina_metrics.(Gauge.inc_one Network.transaction_pool_diff_broadcasted) ;
-  Gossip_net.Any.broadcast_transaction_pool_diff t.gossip_net diff
+  Gossip_net.Any.broadcast_transaction_pool_diff ?nonce t.gossip_net diff
 
-let broadcast_snark_pool_diff t diff =
+let broadcast_snark_pool_diff ?nonce t diff =
```

Comment:
Why is nonce optional?

Reply:
For rebroadcasts we generate a random number for a nonce